spec/rspec/core/world_spec.rb

```diff
@@ -21,6 +21,33 @@ module RSpec::Core
         world.register(example_group)
         expect(world.example_groups).to include(example_group)
       end
+
+      it "invokes before_register callbacks" do
+        world.before_register do |example_group|
+          examples     = example_group.examples
+          new_examples = []
+          examples.each do |ex|
+            caps = [{:browserName => :firefox}, {:browserName => :chrome}]
+            caps.each_with_index do |cap, index|
+              ex_with_cap = ex.clone
+              new_id      = "#{ex_with_cap.id}_#{index}_#{cap}"
+              ex_with_cap.instance_variable_set(:@id, new_id)
+              new_examples << ex_with_cap
+            end
+          end
+
+          example_group.instance_variable_set(:@examples, new_examples)
```

Comment:
An accessor method on example_group for examples would solve that issue.

The only reason I'm mutating IDs is RSpec will not recognize the newly created tests because they have the same id as other tests. These methods are created dynamically so location is not a unique identifier.

metadata is great. [I hit a limitation related to memoization](https://github.com/bootstraponline/sauce_rspec/blob/db6904bfd8fc3b7038f77ad64960ebaef5171a0f/lib/sauce_rspec/rspec.rb#L30) however and I had to monkey patch in an accessor method to bypass this. I updated the spec for this feature to use metadata so users can see a supported way of using the hook.



> Seeing how you intend to use this callback makes me think that we need to provide a different way to solve your problem.

I think this PR gets us partially the way there. I agree it's not a complete solution though.